File: lib/variant/variant.js
'use strict'

/**
 * @classdesc Variant of Diplomacy
 * @memberof variant
 * @prop {!rule.Rule} rule - The rule used in this variant.
 * @prop {!board.Board} initialBoard - The initial state of the board used in this variant.
 */
class Variant {
  /**
   * @constructor
   * @param {!rule.Rule} rule - The rule used in this variant.
   * @param {!board.Board} initialBoard - The initial state of the board used in this variant.
   */
  constructor (rule, initialBoard) {
    this.rule = rule
    this.initialBoard = initialBoard
  }
}

module.exports = Variant

File: core/src/main/scala/diplomacy/variant/Variant.scala
package diplomacy.variant

import diplomacy.board._
import diplomacy.rule.{ Order, Rule }

final case class Variant[State_, Power_ <: Power, MilitaryBranch_ <: MilitaryBranch, UnitStatus_, ProvinceStatus_, Order_ <: Order[Power_, MilitaryBranch_], Result_](
  rule: Rule[State_, Power_, MilitaryBranch_, UnitStatus_, ProvinceStatus_, Order_, Result_],
  initialBoard: Board[State_, Power_, MilitaryBranch_, UnitStatus_, ProvinceStatus_]
)
